### PR TITLE
NEW RECORD TYPE: DNSKEY

### PR DESCRIPTION
**Pull request by xtexChooser**
Add DNSKEY support

### CODE REVIEW

File: models/domain.go [hunk 1]
```diff
@@ -141,7 +141,7 @@ func (dc *DomainConfig) Punycode() error {
 			rec.SetTarget(t)
 		case "CF_REDIRECT", "CF_TEMP_REDIRECT", "CF_WORKER_ROUTE":
 			rec.SetTarget(rec.GetTargetField())
-		case "A", "AAAA", "CAA", "DHCID", "DS", "LOC", "NAPTR", "SOA", "SSHFP", "TXT", "TLSA", "AZURE_ALIAS":
+		case "A", "AAAA", "CAA", "DHCID", "DS", "DNSKEY", "LOC", "NAPTR", "SOA", "SSHFP", "TXT", "TLSA", "AZURE_ALIAS":
```
Comment (tlimoncelli) [inline models/domain.go:144]: Please keep the list alphabetical: DHCID, DNSKEY, DS

File: pkg/js/helpers.js [hunk 2]
```diff
@@ -871,11 +890,11 @@ function IGNORE(labelPattern, rtypePattern, targetPattern) {
 
 // IGNORE_NAME(name, rTypes)
 function IGNORE_NAME(name, rTypes) {
-  return IGNORE(name, rTypes)
+    return IGNORE(name, rTypes);
 }
 
 function IGNORE_TARGET(target, rType) {
-  return IGNORE("*", rType, target)
+    return IGNORE('*', rType, target);
```
Comment (tlimoncelli) [inline pkg/js/helpers.js:897]: Is your IDE rewriting `''` to `""` ?

File: pkg/normalize/validate.go [hunk 3]
```diff
@@ -223,7 +224,7 @@ func checkTargets(rec *models.RecordConfig, domain string) (errs []error) {
 		}
 	case "SRV":
 		check(checkTarget(target))
-	case "CAA", "DHCID", "DS", "IMPORT_TRANSFORM", "SSHFP", "TLSA", "TXT":
+	case "CAA", "DHCID", "DS", "DNSKEY", "IMPORT_TRANSFORM", "SSHFP", "TLSA", "TXT":
```
Comment (tlimoncelli) [inline pkg/normalize/validate.go:227]: Sort: DHCID, DNSKEY, DS

File: integrationTest/integration_test.go [hunk 4]
```diff
@@ -1612,6 +1618,12 @@ func makeTests() []*TestGroup {
 			tc("Create DNAME record in non-FQDN", dname("a", "b")),
 		),
 
+		testgroup("DNSKEY",
+			requires(providers.CanUseDNSKEY),
+			tc("Create DNSKEY record", dnskey("test", 257, 3, 13, "AABBCCDD")),
+			tc("Modify DNSKEY record", dnskey("test", 256, 3, 13, "AABBCCDDEEFF")),
```
Comment (tlimoncelli) [inline integrationTest/integration_test.go:1624]: Please modify each field one at a time (if possible).

Are those values valid for DNSKEY?  (I haven't checked the RFC... they might be valid).  If not, please only use valid examples. Some DNS providers validate the fields and will reject our tests.

**Review by tlimoncelli — COMMENTED** [review]
This looks great so far!  Thank you for taking this on.

This is my first review.  Just a few cosmetic issues.  I'll do a deeper review later.

File: pkg/js/helpers.js (hunk 2 above)
Comment (xtexChooser) [reply pkg/js/helpers.js:897]: those are done by prettier (`v3.2.5`)

**Comment by tlimoncelli** [comment]
Fantastic!  Thank you so much for contributing DNSKEY to the project!  I'm sure a lot of people will find it useful!

**Pull request merged**